mlir/include/mlir/Dialect/Linalg/IR/Linalg.h

```diff
@@ -144,4 +144,126 @@ std::pair<int64_t, int64_t> getFmrFromWinogradConv2DFmr(WinogradConv2DFmr fmr);
 #define GET_OP_CLASSES
 #include "mlir/Dialect/Linalg/IR/LinalgRelayoutOps.h.inc"
 
+namespace mlir::linalg {
+
+/// Specialization of `linalg.matmul` op that has a transpose map on A
+class MatmulTransposeAOp : public MatmulOp {
```

Comment:
Should we immediately mark this as deprecated with `LLVM_DEPRECATED`? Same for the other transition classes below. IIUC, the intention is for this to be short-lived.